internal/cmd/cli/apply/apply.go

```diff
@@ -736,3 +741,18 @@ func deleteSecretIfExists(ctx context.Context, c client.Client, name, ns string)
 
 	return nil
 }
+
+func sendWarningEvent(r record.EventRecorder, namespace, artifactID string, errorToLog error) {
+	jobName := os.Getenv("JOB_NAME")
+	if jobName == "" {
+		logrus.Warn("JOB_NAME environment variable not set")
+		return
+	}
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: namespace,
+		},
+	}
+	r.Event(job, fleetevent.Warning, "Failed", fmt.Sprintf("deleting OCI artifact: %q, error: %v", artifactID, errorToLog.Error()))
```

Comment:
Should we be a bit more explicit about the reason for the event?
e.g.:
```suggestion
	r.Event(job, fleetevent.Warning, "FailedToDeleteOCIArtifact", fmt.Sprintf("deleting OCI artifact %q: %v", artifactID, errorToLog.Error()))
```